libs/plugin-api/src/main/java/org/elasticsearch/plugin/api/settings/Inject.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugin.api.settings;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation used to mark constructor to inject settings instance.
+ * A constructor parameter has to be an interface marked with appropriate annotation (i.e AnalysisSetting)
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.CONSTRUCTOR)
+public @interface Inject {
```

Comment:
I'm not fully up to date with all discussions, but I'm wondering if we considered maybe using another term than Inject? It might be confusing to plugin developers that are using their own dependency injection frameworks.

Reply:
I agree, it can be confusing for elasticserach developers too. We already have `org.elasticsearch.common.inject.Inject`.
Any ideas for better name? `@SettingsInject` `@WithSettings`?

Reply:
I like `@InjectSettings` - will update